### PR TITLE
✨Destination Iceberg: Bump Iceberg from 1.1.0 to 1.3.0 and add REST catalog support

### PR DESCRIPTION
**Pull request by nastra**
## What
*This updates the Iceberg dependency from 1.1.0 to 1.3.0 and also adds REST catalog support*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

**Comment by nastra** [comment]
image

File: airbyte-integrations/connectors/destination-iceberg/build.gradle [hunk 1]
```diff
@@ -14,24 +14,24 @@ dependencies {
...
-    implementation "software.amazon.awssdk:utils:2.17.257"
-    implementation "software.amazon.awssdk:url-connection-client:2.17.257"
-    implementation "software.amazon.awssdk:s3:2.17.257"
-    implementation "software.amazon.awssdk:glue:2.17.257"
-    implementation "software.amazon.awssdk:dynamodb:2.17.257"
-    implementation "software.amazon.awssdk:kms:2.17.257"
-    implementation "software.amazon.awssdk:sts:2.17.257"
-    implementation "software.amazon.awssdk:sdk-core:2.17.257"
-    implementation "software.amazon.awssdk:aws-core:2.17.257"
+    implementation "software.amazon.awssdk:utils:2.20.18"
```
Comment (nastra) [inline airbyte-integrations/connectors/destination-iceberg/build.gradle:26]: this is to match the AWS SDK version being used by Iceberg: https://github.com/apache/iceberg/blob/apache-iceberg-1.3.0/versions.props#L24

File: airbyte-integrations/connectors/destination-iceberg/src/main/java/io/airbyte/integrations/destination/iceberg/IcebergConstants.java [hunk 2]
```diff
@@ -27,6 +27,9 @@ public class IcebergConstants {
   public static final String JDBC_PASSWORD_CONFIG_KEY = "password";
   public static final String JDBC_SSL_CONFIG_KEY = "ssl";
   public static final String JDBC_CATALOG_SCHEMA_CONFIG_KEY = "catalog_schema";
+  public static final String REST_CATALOG_URI_CONFIG_KEY = "rest_uri";
+  public static final String REST_CATALOG_CREDENTIAL_CONFIG_KEY = "rest_credential";
```
Comment (nastra) [inline airbyte-integrations/connectors/destination-iceberg/src/main/java/io/airbyte/integrations/destination/iceberg/IcebergConstants.java:31]: REST catalog uses either a token or a credential to interact with the REST server: https://github.com/apache/iceberg/blob/apache-iceberg-1.3.0/core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Properties.java#L24-L28

File: airbyte-integrations/connectors/destination-iceberg/src/test-integration/resources/rest-catalog-compose.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+version: "3"
+
+services:
```
Comment (nastra) [inline airbyte-integrations/connectors/destination-iceberg/src/test-integration/resources/rest-catalog-compose.yml:3]: This is a slimmed down version of Iceberg's quickstart example: https://iceberg.apache.org/spark-quickstart/

**Comment by marcosmarxm** [comment]
Hello @nastra I added this our team's backlog and I'm going to review during the week.

**Review by marcosmarxm — APPROVED** [review]
Thanks @nastra

**Comment by octavia-approvington** [comment]
This looks fine!  <br/> Merged! <br/> Imagine it being fine

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**